src/QueryBuilder/ResultViewer/ResultViewer.js

```diff
@@ -0,0 +1,119 @@
+import React, { memo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Col, Row, Accordion, AccordionSet, MultiColumnList, Headline } from '@folio/stripes/components';
+import { PrevNextPagination, usePagination } from '@folio/stripes-acq-components';
+import { QueryLoader } from './QueryLoader';
+import { getTableMetadata } from './helpers';
+
+export const ResultViewer = memo(({
+  showPagination = true,
+  contentData = [],
+  limit = 100,
+  offset = 0,
+  onPageChange,
+  entityType,
+  headline,
+  headlineEnd,
+  visibleColumns,
+  onSetDefaultVisibleColumns,
+  onSetDefaultColumns,
+  accordionHeadline,
+  isInProgress,
+  inProgressTitle,
+  height,
+  mlcLoading,
+}) => {
+  const { changePage, pagination } = usePagination({ limit, offset });
+  const { columnMapping, defaultColumns, defaultVisibleColumns } = () => getTableMetadata(entityType);
+
+  useEffect(() => {
+    onPageChange?.(pagination);
+  }, [pagination]);
+
+  // set visible by default columns once
+  useEffect(() => {
+    if (defaultVisibleColumns && defaultColumns) {
+      onSetDefaultColumns?.(defaultColumns);
+      onSetDefaultVisibleColumns?.(defaultVisibleColumns);
+    }
+  }, [defaultVisibleColumns, defaultColumns]);
+
+  const renderTable = () => (
+    <>
+      <Row between="xs">
+        <Col xs={6}>
+          <Headline size="large" margin="none" tag="h3">
+            {headline}
+          </Headline>
+        </Col>
+        {headlineEnd}
+      </Row>
+      <Row>
+        <Col xs={12}>
+          <MultiColumnList
+            contentData={contentData}
+            columnMapping={columnMapping}
+            visibleColumns={visibleColumns}
+            pagingType={null}
+            onNeedMoreData={changePage}
+            height={height}
+            loading={mlcLoading}
+          />
+          {showPagination && (
+            <PrevNextPagination
+              {...pagination}
+              totalCount={contentData?.length}
+              onChange={changePage}
+            />
+          )}
+        </Col>
+      </Row>
+    </>
+  );
+
+  const renderWithAccordion = () => (
+    <AccordionSet>
+      <Accordion
```

Comment:
I even don't understand why we need accordion here, when its name comes from outside

now we do if sections just to support cases when viewer is rendered with accordion or without

Reply:
For now we should support 3 potential places where ResultsViewer can be used.
- As a part of query-builder plugin (accordion is not required.)
- Called in bulk edit (accordion is required)
- Called in lists app (accordion is required)

We can simplify it, if design will be unified a little bit more across the places.